efi/image_rules.go

```diff
@@ -0,0 +1,33 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package efi
+
+import "fmt"
+
+type (
+	newImageLoadHandlerFn func(imageRuleSet, peImageHandle) (imageLoadHandler, error)
```

Comment:
this type is not really used yet except as a return type, will there be something that consumes it?

Reply:
and maybe as remarked its name need to be new...OnRules?

Reply:
> this type is not really used yet except as a return type, will there be something that consumes it?

It will be used in the next couple of PRs.

Reply:
I've changed this PR a bit to remove this type - it will get added back in a subsequent PR, although it will look a bit different.